rofl-client/rs/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+    }
+    if let Some(b) = body {
+        req.extend_from_slice(format!("Content-Length: {}\r\n", b.len()).as_bytes());
+    }
+    req.extend_from_slice(b"\r\n");
+    if let Some(b) = body {
+        req.extend_from_slice(b);
+    }
+
+    stream.write_all(&req)?;
+    stream.flush()?;
+
+    let mut resp = Vec::new();
+    let mut buf = [0u8; 8192];
+    loop {
+        let n = stream.read(&mut buf)?;
+        if n == 0 {
+            break;
+        }
+        resp.extend_from_slice(&buf[..n]);
+    }
+
+    let header_end = resp
+        .windows(4)
+        .position(|w| w == b"\r\n\r\n")
+        .ok_or_else(|| {
+            Error::new(
+                ErrorKind::InvalidData,
+                "Invalid HTTP response: no header/body delimiter",
+            )
+        })?;
+    let (head, body_bytes) = resp.split_at(header_end + 4);
+
+    let mut lines = head.split(|&b| b == b'\n');
+    let status_line = lines
+        .next()
+        .ok_or_else(|| Error::new(ErrorKind::InvalidData, "Invalid HTTP response: empty"))?;
+    let status_str = String::from_utf8(status_line.to_vec())
+        .map_err(|e| Error::new(ErrorKind::InvalidData, e))?;
+    let code: u16 = status_str
+        .split_whitespace()
+        .nth(1)
+        .ok_or_else(|| Error::new(ErrorKind::InvalidData, "Invalid HTTP status line"))?
+        .parse()
+        .map_err(|e| Error::new(ErrorKind::InvalidData, e))?;
+
+    if !(200..300).contains(&code) {
+        let msg = String::from_utf8_lossy(body_bytes).to_string();
+        return Err(Error::other(format!("HTTP {code} error: {msg}")));
+    }
+
+    Ok(body_bytes.to_vec())
+}
+
+// See https://github.com/oasisprotocol/oasis-sdk/blob/1ae8882b05d10a44398e52b5b8c56ab2086f81b3/rofl-appd/src/services/kms.rs#L59-L74
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum KeyKind {
+    Raw256,
+    Raw384,
+    Ed25519,
+    Secp256k1,
+}
+
+impl std::fmt::Display for KeyKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            KeyKind::Raw256 => write!(f, "raw-256"),
+            KeyKind::Raw384 => write!(f, "raw-384"),
+            KeyKind::Ed25519 => write!(f, "ed25519"),
+            KeyKind::Secp256k1 => write!(f, "secp256k1"),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct KeyGenerationRequest {
+    key_id: String,
+    kind: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct KeyGenerationResponse {
+    key: String,
+}
+
+// -------------------- sign-submit types --------------------
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "kind", content = "data")]
+pub enum Tx {
+    #[serde(rename = "eth")]
+    Eth(EthCall),
+    #[serde(rename = "std")]
+    Std(String), // CBOR-serialized hex-encoded Transaction
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EthCall {
+    pub gas_limit: u64,
+    pub to: String,
+    pub value: u64,
```

Comment:
Consider changing to `String` for consistency with Python/TS clients, which normalize all values to decimal strings to support amounts > u64::MAX. For context https://github.com/oasisprotocol/oasis-sdk/pull/2416#discussion_r2627631517